### ISSUE TITLE
Index was outside the bounds of the array

### ISSUE THREAD

**Issue by tiperes** [linq2db/linq2db#2854]
I am a Tech Lead in large software factory and our team has been using linq2db in the last 2 years.
We use System.Linq.Dynamic.Core library to build dynamically LINQ queries at runtime and linq2db as the LINQ provider for querying an Oracle database.

In the last core components revision, we have tested the upgrade of linqtodb from version 3.2.1 to version 3.2.3.
All applications depending on this library stopped working.
We have reverted back to linqtodb version 3.2.1.

This is not critical for us, at least for now, but we would like to continue to receive updates from this library.
Feel free to request any additional detail that you need from me.

Thanks!

**Exception message:**
   Index was outside the bounds of the array.

**Stack trace:**
```
   at LinqToDB.DataProvider.Oracle.Oracle11SqlBuilder.IsValidIdentifier(String name)
   at LinqToDB.DataProvider.Oracle.Oracle11SqlBuilder.Convert(StringBuilder sb, String value, ConvertType convertType)
   at LinqToDB.SqlProvider.BasicSqlBuilder.ConvertInline(String value, ConvertType convertType)
   at LinqToDB.SqlProvider.BasicSqlBuilder.GetPhysicalTableName(ISqlTableSource table, String alias, Boolean ignoreTableExpression, String defaultDatabaseName)
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildPhysicalTable(ISqlTableSource table, String alias, String defaultDatabaseName)
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildTableName(SqlTableSource ts, Boolean buildName, Boolean buildAlias)
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildFromClause(SqlStatement statement, SelectQuery selectQuery)
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildSelectQuery(SqlSelectStatement selectStatement)
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildSql(Int32 commandNumber, SqlStatement statement, StringBuilder sb, OptimizationContext optimizationContext, Int32 indent, Boolean skipAlias)
   at LinqToDB.SqlProvider.BasicSqlBuilder.BuildSql(Int32 commandNumber, SqlStatement statement, StringBuilder sb, OptimizationContext optimizationContext, Int32 startIndent)
   at LinqToDB.Data.DataConnection.QueryRunner.GetCommand(DataConnection dataConnection, IQueryContext query, IReadOnlyParameterValues parameterValues, Int32 startIndent)
   at LinqToDB.Data.DataConnection.QueryRunner.CreateExecutionQuery(DataConnection dataConnection, IQueryContext context, IReadOnlyParameterValues parameterValues)
   at LinqToDB.Data.DataConnection.QueryRunner.SetQuery(IReadOnlyParameterValues parameterValues)
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReader()
   at LinqToDB.Linq.QueryRunner.<ExecuteQuery>d__12`1.MoveNext()
```

### Steps to reproduce
Any Linq query (example: "select * from TABLE") using System.Linq.Dynamic.Core library and linqtodb as the database provider against an Oracle 11g database.

### Environment details
LINQ library: System.Linq.Dynamic.Core 1.27 (https://github.com/zzzprojects/System.Linq.Dynamic.Core/releases/tag/v1.2.7)
linq2db version: 3.2.3 (https://github.com/linq2db/linq2db/releases/tag/v3.2.3)
Database Server: Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
Database Provider: Oracle Data Provider for .NET, Managed Driver 4.122.1.20170524
Operating system: Windows Server 2012 R2
.NET Framework: C# .NET Framework 4.7.2

**Comment by jogibear9988** [comment]
I've the same issue (I think). I've created a test #2855

**Comment by jogibear9988** [comment]
Can you post the exact query you'd like to do? And the Classes.

**Comment by sdanyliv** [comment]
@tiperes. Please share your model and how do you use Dynamic LINQ.

**Comment by MaceWindu** [comment]
Checked code. It happens due to empty string (`""`) specified as table alias for table expression.

For better fix we must understand source of this empty alias, so could you provide more details what kind of table expression used in your code?

**Comment by tiperes** [comment]
I'm isolating a use case, and I'll share with you until the end of the week.
Thanks!

**Comment by tiperes** [comment]
Hello!

First of all, sorry for the delay in the response. It was hard to understand what was going on.
After isolating one of the use cases, the issue stopped to occur... :|
I had to try more variations, recheck against our 2 databases versions, all of this toggling between the 2 linq2db versions.

The problem (or our problem) was is in the database schema name, not in the table alias.
Most of the scenarios where we use Dynamic Linq and Linq2db approach, are based upon the usage of dynamic entities built at runtime (so not pre-declared/pre-compiled).
We do this at runtime using a generic algorithm we've developed, and within this process we were applying a rule based on the scope of the request to determine if the entity/table schema was needed to be fill. If it was a table from the same schema as the user executing the request, we were not filling it, so we were setting the table schema with an empty string...

So in reality this was not an issue from linq2db library. This was related to an issue we've injected in our framework recently and the related apps are not yet using it enough, so it was not detected before.

Sorry for all the trouble and thank for your quick reaction and feedback.
Continue the good work! Many Thanks!
Best Regards!
